libcontainer/seccomp/patchbpf/enosys_linux.go

```diff
@@ -523,6 +523,10 @@ func assemble(program []bpf.Instruction) ([]unix.SockFilter, error) {
 }
 
 func generatePatch(config *configs.Seccomp) ([]bpf.Instruction, error) {
+	// Path the generated cBPF only when there is not a defaultErrnoRet set
+	if config.DefaultErrnoRet != nil {
+		return nil, nil
+	}
```

Comment:
This behaviour is incorrect -- and is what I meant when I said a while ago in the runtime-spec that `defaultErrnoRet` is *useful* but doesn't solve the problem that the patching does.

We must *always* patch the cBPF program if the action is a "deny" action. The fact that the errno is not `-EINVAL` isn't really a good enough reason to re-enable the seccomp footgun (someone is going to set it to `-EACCES` and that'll be a ticking timebomb for no good reason). Maybe we could make it so that `-ENOSYS` is special-cased but that that's the only exception.

Reply:
fair.  I'd expect `ENOSYS` to be the only reasonable value for `DefaultErrnoRet` anyway.  I've adjusted the patch to skip patching only when `DefaultErrnoRet == ENOSYS`

Reply:
Thanks! Yeah I would hope *most* people wouldn't set it to anything else, but you can never be too careful.